### PR TITLE
Improvements for cmake

### PR DESCRIPTION
**Pull request by Dead2**
- Move cmake helpers to separate folder
- Improve compatibility for using Stabilizer as a subproject
- Various cleanup and improvements to CMake scripts

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.21)
 # 3.21 - CMAKE_<LANG>_LINKER_LAUNCHER
 
 project(stabilizer CXX)
+enable_language(C)
```
Comment (magras) [inline CMakeLists.txt:5]: I'd like to keep the reason why we enable plain C as commentary:
> CRAP: we don't need C language, but `find_package(LLVM)` forces the check

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,11 +12,73 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 include(GNUInstallDirs)
 
-add_subdirectory(pass)
-add_subdirectory(runtime)
+
+### LLVMStabilizer.so
```
Comment (magras) [inline CMakeLists.txt]: With `.so` suffix present, I would add `lib` prefix.

File: CMakeLists.txt [hunk 3]
```diff
@@ -11,11 +12,73 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
+add_library(stabilizer MODULE)
+
+target_include_directories(stabilizer
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/Heap-Layers
+    ${CMAKE_CURRENT_SOURCE_DIR}/DieHard/src/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/DieHard/src/include/math
+    ${CMAKE_CURRENT_SOURCE_DIR}/DieHard/src/include/rng
+    ${CMAKE_CURRENT_SOURCE_DIR}/DieHard/src/include/static
+    ${CMAKE_CURRENT_SOURCE_DIR}/DieHard/src/include/util
```
Comment (magras) [inline CMakeLists.txt]: I believe this paths are relative to `CMAKE_CURRENT_SOURCE_DIR` by default, so it is redundant after the move.

File: CMakeLists.txt [hunk 4]
```diff
@@ -35,23 +99,23 @@ if (BUILD_TESTING)
...
+    INPUT ${CMAKE_CURRENT_BINARY_DIR}/cmake/StabilizerToolchain.cmake.tmp
   )
 
   # Add tests project
   include(ExternalProject)
   ExternalProject_Add(
     tests
     DEPENDS LLVMStabilizer stabilizer
-    SOURCE_DIR ${CMAKE_SOURCE_DIR}/tests
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/tests
```
Comment (magras) [inline CMakeLists.txt:117]: I think we can omit `CMAKE_CURRENT_SOURCE_DIR` here too.

File: CMakeLists.txt (hunk 2 above)
Comment (Dead2) [reply CMakeLists.txt]: Good catch

File: CMakeLists.txt (hunk 1 above)
Comment (Dead2) [reply CMakeLists.txt:5]: I'll add it back.

File: CMakeLists.txt (hunk 3 above)
Comment (Dead2) [reply CMakeLists.txt]: Yeah, no need for it here.

File: CMakeLists.txt (hunk 4 above)
Comment (Dead2) [reply CMakeLists.txt:117]: I am more tempted to keep this for consistency actually, even if it does not alter the behavior.

File: CMakeLists.txt (hunk 1 above)
Comment (Dead2) [reply CMakeLists.txt:5]: Added back a modified comment

**Comment by Dead2** [comment]
@magras Updated with most of the requested changes now.
Have you tested running this in your environment? I have only tested compiling normally, have not tested docker or anything else.

File: CMakeLists.txt (hunk 4 above)
Comment (magras) [reply CMakeLists.txt:117]: OK, I'm fine with that.

**Comment by nmoinvaz** [comment]
I'm not seeing those updates. Other than that it looks good.

**Comment by magras** [comment]
@Dead2, for me it is necessary to merge #26, but otherwise it works.

**Comment by Dead2** [comment]
> I'm not seeing those updates. Other than that it looks good.

I forgot to push the changes, fixed now 👍

**Review by magras — APPROVED** [review]
Looks good to me.

**Comment by Dead2** [comment]
> @Dead2, for me it is necessary to merge #26, but otherwise it works.

I am working on a few changes to hopefully fulfill the needs of #9 and #26 👍

**Pull request merged**